### ISSUE TITLE
Need to delete cloudflare DNSs everytime before i restart the addon

### ISSUE THREAD

**Issue by UCChristianJones** [brenner-tobias/addon-cloudflared#259]
### The problem

I get the following in the logs after each restart (prior to deleting the sub domains on cloudflare):

s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service init-banner: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
-----------------------------------------------------------
 Add-on: Cloudflared
 Use a Cloudflare Tunnel to remotely connect to Home Assistant without opening any ports
-----------------------------------------------------------
 Add-on version: 4.0.4
 You are running the latest version of this add-on.
 System: Home Assistant OS 9.3  (aarch64 / raspberrypi4-64)
 Home Assistant Core: 2022.11.4
 Home Assistant Supervisor: 2022.10.2
-----------------------------------------------------------
 Please, share the above information when looking for help
 or support in, e.g., GitHub or forums.
-----------------------------------------------------------
s6-rc: info: service init-banner successfully started
s6-rc: info: service init-log-level: starting
s6-rc: info: service init-log-level successfully started
s6-rc: info: service init-cloudflared-config: starting
[09:38:03] INFO: Checking config for legacy options...
[09:38:04] INFO: Checking add-on config...
[09:38:06] INFO: Checking for existing certificate...
[09:38:06] INFO: Existing certificate found
[09:38:06] INFO: Checking for existing tunnel...
[09:38:06] INFO: Existing tunnel with ID e381b62e-275a-4707-afa4-1017124a5ef3 found
[09:38:06] INFO: Checking if existing tunnel matches name given in config
[09:38:07] INFO: Existing Cloudflare Tunnel name matches config, proceeding with existing tunnel file
[09:38:07] INFO: Creating config file...
[09:38:09] INFO: Validating config file...
Validating rules from /tmp/config.json
OK
[09:38:09] INFO: Creating DNS entry ###.###.tk...
Failed to add route: code: 1003, reason: An A, AAAA, or CNAME record with that host already exists.
[09:38:11] FATAL: Failed to create DNS entry ###.###.tk.
s6-rc: warning: unable to start service init-cloudflared-config: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
prog: fatal: stopping the container.
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service init-log-level: stopping
s6-rc: info: service init-log-level successfully stopped
s6-rc: info: service init-banner: stopping
s6-rc: info: service init-banner successfully stopped
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped

If the sub domain already exists, the addon should continue I would have thought, otherwise it means, I cannot access home assistant externally if the device restarts, which is the point of this addon..

### What version of Cloudflared has the issue?

4.0.4

### What type of installation are you running?

Home Assistant OS

### Steps to reproduce the issue

Setup the addon, start it and make sure it all works and is accessible externally, then restart the addon - the pasted log is what is shown and I am not able to access my HASSIO.

**Comment by zilchulant** [comment]
Current workaround for me: uninstall and reinstall the addon.

**Comment by brenner-tobias** [comment]
This is indeed very strange. What the add-on does, is override the CNAME record that you define as external_hostname with the CNAME to the tunnel. Now the only thing I can think of is that you have another record (A or AAAA) in there that is "blocking" the creation of the CNAME.
Now the strange part is that you obviously delete all the entries, otherwise the add-on would not be able to start.
Therefor, it would be great if you had a look at your Cloudflare DNS Dashboard when the add-on is running (or stopped after running and when you cannot start). Is there any other entry than the CNAME to the tunnel?

**Comment by UCChristianJones** [comment]
I've fixed it. But I had to setup and new/clean copy of home assistant. 
The only thing I could think this could have been, is I had one URL first on cloudflare (which was used in cloudflared), then I switched across to a new URL, and cloudflared failed due to having the same tunnel name (which had previously been used for the other URL). Then I changed the tunnel name in cloudflared, and that worked. But when I restarted the addon, its like cloudflared wasn't checking to see if the CNAME had been setup previously. So not sure if that would give you something to go on. But I would recommend closing this for now, and wait and see if someone else gets this again in the future, and are able to be more detailed in their steps/reproduction.
